app/controllers/rounds_controller.rb

```diff
@@ -142,59 +193,85 @@ def round_params
+    return '-' if score1 == 0 && score2 == 0 # rubocop:disable Style/NumericPredicate
 
-  def pairing_player2(stage, pairing)
-    pairing_player(stage, pairing.player2, pairing.player2_side)
-  end
+    ws = winning_side(score1_corp, score1_runner, score2_corp, score2_runner)
 
-  def pairing_player(stage, player, side)
-    {
-      name_with_pronouns: player.name_with_pronouns,
-      side:,
-      side_label: side_label(stage, side),
-      corp_id: pairing_identity(player.corp_identity_object),
-      runner_id: pairing_identity(player.runner_identity_object)
-    }
+    return "#{score1} - #{score2}" unless ws
+
+    "#{score1} - #{score2} (#{ws})"
   end
 
-  def side_label(stage, side)
-    return nil unless stage.single_sided? && side
+  def winning_side(score1_corp, score1_runner, score2_corp, score2_runner)
+    corp_score = (score1_corp || 0) + (score2_corp || 0)
+    runner_score = (score1_runner || 0) + (score2_runner || 0)
 
-    "(#{side.to_s.titleize})"
+    if (corp_score - runner_score).zero?
+      nil
+    elsif (corp_score - runner_score).negative?
+      'R'
+    else
+      'C'
+    end
   end
 
-  def pairing_identity(identity)
-    return nil unless identity
+  def name_with_pronouns(player)
+    if player.nil?
+      '(Bye)'
+    else
+      !player['pronouns'].empty? ? "#{player['name']} (#{player['pronouns']})" : player['name']
+    end
+  end
 
-    {
-      name: identity.name,
-      faction: identity.faction
-    }
+  def corp_id(player)
+    id(player, 'corp')
   end
 
-  def score_label(pairing)
-    return '-' if pairing.score1 == 0 && pairing.score2 == 0 # rubocop:disable Style/NumericPredicate
+  def runner_id(player)
+    id(player, 'runner')
+  end
 
-    ws = winning_side(pairing)
+  def id(player, side)
+    if player.nil?
+      nil
+    else
+      {
+        "name": player["#{side}_identity"],
+        "faction": player["#{side}_faction"]
+      }
+    end
+  end
 
-    return "#{pairing.score1} - #{pairing.score2}" unless ws
+  def player1_side(side)
+    if side.nil?
+      nil
+    else
+      (side == 'player1_is_corp' ? 'corp' : 'runner')
+    end
+  end
 
-    "#{pairing.score1} - #{pairing.score2} (#{ws})"
+  def player1_side_label(side)
+    if side.nil?
+      nil
+    else
+      "(#{(side == 'player1_is_corp' ? 'corp' : 'runner').to_s.titleize})"
+    end
   end
 
-  def winning_side(pairing)
-    corp_score = (pairing.score1_corp || 0) + (pairing.score2_corp || 0)
-    runner_score = (pairing.score1_runner || 0) + (pairing.score2_runner || 0)
+  def player2_side(side)
+    if side.nil?
+      nil
+    else
+      (side == 'player1_is_corp' ? 'runner' : 'corp')
+    end
+  end
 
-    if (corp_score - runner_score).zero?
+  def player2_side_label(side)
+    if side.nil?
       nil
-    elsif (corp_score - runner_score).negative?
-      'R'
     else
-      'C'
+      "(#{(side == 'player1_is_corp' ? 'runner' : 'corp').to_s.titleize})"
     end
   end
```

Comment:
For all 4 of these methods, I'm a bit confused because the field is an integer in the database. How is Rails mapping it to a string during the pluck?

Please can we add coverage for this in the tests in rounds_controller_spec.rb? Right now the tests are all using players with no IDs set, and in double elim there's no side set for any of the pairings. We'll need tests with both of those populated with data.